### PR TITLE
refactor: lift worker-process + event-log substrate out of src/acp

### LINKED ISSUE #2092: Plugins 02: lift subprocess substrate to src/process/ + event bus to src/events/
**Note:** Claude handled the investigation and prose via back-and-forth. Idea, decisions, and everything else are mine. — @Seluj78

Part of #268; design in `docs/development/internals/plugin-system.md` (D1, D2).

Untouched by the minimal core (#2311): the substrate lift is still entirely ahead of us.

## Scope

Pure substrate lift, zero behavior change:

- Move the protocol-agnostic subprocess machinery (`src/acp/runner.rs`, `src/acp/worker_registry.rs`, `src/acp/supervisor.rs`) to a neutral `src/process/` module. Both `src/acp/` and the future plugin host consume it; ACP must never depend on plugin-host concepts (the dependency arrow points from both consumers down to `src/process/`).
- Generalize the event-store triple (durable SQLite seq log + best-effort broadcast + replay-from-seq, today in `src/acp/event_store.rs`) into `src/events/` as the Core Event Bus, keyed by topic. Publication only: writes go through core services which then publish post-mutation facts. ACP becomes a consumer.
- High-volume pane snapshots stay out of the durable log.

## Acceptance

- No behavior change: existing ACP e2e and unit tests pass unmodified.
- `src/acp/` contains only ACP protocol semantics after the lift.

Depends on nothing; unblocks #2095 (the Tier 1 host).

### PR DESCRIPTION
**Pull request by Seluj78**
**Note:** Claude handled the implementation and prose via back-and-forth. Idea, decisions, and everything else are mine. — @Seluj78

## Description

Plugins 02 (part of #268, follows the minimal core in #2311): lift the protocol-agnostic substrate out of `src/acp/` so the future plugin host (#2095) can reuse it without depending on ACP. The dependency arrow now runs `acp -> {process, events}`, never the reverse.

This is a pure substrate lift with **zero behavior change**. Scoped per discussion to a true decomposition (not a whole-file move), touching the worker_registry utilities only and leaving `runner.rs` / `supervisor.rs` in place as consumers.

**Part A, event bus -> `src/events/`.** A new generic event-log storage core: `Schema` (validated table prefix), `open`, `insert_event`, per-topic `prune_retention` with caller-supplied pinned discriminants, keyset `scan` (`SeqBound` + `Order`), seq bookkeeping, attachment blob storage, topic deletion, and an activity-time query with caller-supplied exclusions. It is keyed by an opaque `topic` over opaque JSON payloads and knows nothing about ACP. `acp::event_store::EventStore` keeps its byte-identical public API (it still owns the `Connection`, so existing tests that reach `store.conn` pass unmodified) and delegates the generic mechanics to `crate::events`, keeping the ACP-specific layer (Event serialization, the replay has-more / cursor / turn-boundary logic, and the `json_extract` accessors) behind. `Schema::new("acp")` maps to the existing `acp_events` / `acp_attachments` tables, so there is **no data migration**.

**Part B, worker process utils -> `src/process/`.** A new `src/process/worker.rs`: process-group signalling (terminate/kill/reap), pid liveness, the runner self-inspection state machine, and the `<dir>/<id>.{json,sock,log,restart}` path builders. None of it knows about ACP: the consumer supplies the base directory and a pid extractor for record inspection, which preserves the exact `Unreadable`-on-parse-failure watchdog semantics. `acp::worker_registry` keeps the ACP `WorkerRecord` and its persistence and now delegates the generic plumbing.

A couple of notes on the issue text, confirmed against the code during investigation: the "best-effort broadcast" the issue places in `event_store.rs` actually lives in `supervisor::ChannelSink`, so generalizing it ships with the deferred supervisor decomposition, not here; and `runner.rs` / `supervisor.rs` are mostly ACP-specific, so only the worker_registry utilities and the event log actually lift.

Closes #2092

## PR Type

- [ ] New Feature
- [x] Refactor
- [ ] Bug Fix
- [ ] Documentation
- [ ] Infrastructure / CI

## How to test

Behavioral checks that should all behave exactly as on `main` (the point of the refactor is that nothing changes):

- [ ] Start `aoe serve`, open a structured-view session, run a turn, then reload the page: the transcript replays from the event store as before.
- [ ] Scroll up in a long structured-view session: older history pages in (recent-first) aligned to user-turn boundaries, with no seamed/split turns.
- [ ] Restart the daemon mid-session and reattach: the transcript rehydrates and seq counters resume without colliding.
- [ ] Attach a screenshot to a prompt, reload, and fetch it: the attachment still loads; deleting the session leaves no orphaned blobs.
- [ ] `aoe acp stop|kill|restart <session>`: the worker and its child tree are reaped (no orphaned `node` / agent processes), and restart respawns on the next reconciler tick.
- [ ] Leave a session idle past `acp.auto_stop_idle_secs`: the idle-reap still fires (lifecycle/metadata events do not reset the idle clock).

## Checklist

- [x] New and existing tests pass
- [x] Documentation was updated where necessary
- [ ] For UI changes: included screenshot or recording

## Test Coverage Analysis

**Web dashboard / structured view (Playwright user story)**
- [x] N/A: this PR doesn't change a user-facing dashboard flow

**TUI / CLI (e2e test)**
- [x] N/A: this PR doesn't change TUI rendering, a CLI subcommand, or session lifecycle

Behavior is unchanged, so the existing ACP e2e / integration tests pass unmodified and remain the regression net. New unit tests cover the extracted substrate directly: `src/events` is driven with a non-ACP table prefix and two topics (append / scan / retention / attachments / activity-time) to prove it is genuinely reusable, and `src/process/worker` covers the parameterized path builders, `ensure_dir` permissions, and the pid-extractor inspection state machine.

## AI Usage

- [ ] No AI was used
- [x] AI was used

**AI Model/Tool used:** Claude (via Claude Code, agent-of-empires `/aoe-implement` skill)

**Any Additional AI Details you'd like to share:**
Investigation, design (pressure-tested with a multi-model debate), and implementation were drafted by Claude under my direction. I made the scope calls (decompose vs move; worker_registry utils only; one PR with staged commits), reviewed each commit, and ran the validation.

- [x] I am an AI Agent filling out this form (check box if true)

---
<a href="https://app.blacksmith.sh/agent-of-empires/codesmith/agent-of-empires/pr/2374"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1784925596&installation_id=139138837&pr_number=2374&repository=agent-of-empires%2Fagent-of-empires&return_to=https%3A%2F%2Fgithub.com%2Fagent-of-empires%2Fagent-of-empires%2Fpull%2F2374&signature=aee9f8b6571b232c4d377a9a9698f92a1c5d627c4dbe0859dc3a3cf389003af7"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

This change lifts ACP-specific infrastructure into shared, protocol-neutral modules so it can be reused by a future plugin host without pulling in ACP-specific dependencies, while keeping existing behavior intact.

## What changed
- Added a new shared durable event-log core in `src/events/` (topic-keyed SQLite storage) with:
  - idempotent event append + bounded replay/scan
  - per-topic sequence bookkeeping
  - retention pruning with “pinned/non-substantive” exclusions
  - durable attachment blob support and topic deletion
- Added a new shared supervised worker substrate in `src/process/worker.rs` with:
  - PID liveness checks
  - process-group termination/kill/escalation
  - runner record inspection/state classification
  - validated path builders for worker artifacts
- Refactored ACP to delegate the generic parts to these shared modules:
  - `acp::event_store::EventStore` keeps its public API but uses `crate::events` for the durable storage mechanics (while retaining ACP-specific serialization/replay logic).
  - `acp::worker_registry` keeps ACP record persistence responsibilities, but uses `crate::process::worker` for the shared process/subprocess plumbing.
- Updated ACP shutdown/watchdog/CLI/reconciler code paths to consistently use the shared worker process-group helpers.
- Updated docs and repo guidance to document the new “shared substrate” modules and their intended reuse.
- Kept everything feature-gated under `serve` for the server runtime.

## What moved
- Durable event-log mechanics: `src/acp/` → `src/events/`
- Worker subprocess/process-group substrate: `src/acp/` → `src/process/worker.rs`
- ACP remains the “consumer” for both durable events and worker supervision, preserving existing table naming via an ACP schema prefix.

## Benefits
- Clear separation between “protocol semantics” (ACP) and reusable infrastructure (durable event-log + supervised worker plumbing).
- Less duplication and a cleaner foundation for adding the plugin host while reusing the same durability and worker-lifecycle primitives.
- No migration expected: ACP continues using the same underlying table naming strategy, and ACP tests are intended to pass unchanged.
- Retention correctness: pruning now deletes eligible event rows and cleans up attachments in a way that preserves blobs for pinned events that should survive pruning.

## Potential follow-ups
- Validate (via targeted tests if needed) the retention edge cases around “pinned” items and attachment lifecycles across prune failures and paging/replay boundaries, especially as the plugin host starts using the shared substrate.

### CODE REVIEW

File: src/events/mod.rs [hunk 1]
```diff
@@ -0,0 +1,678 @@
...
+/// Build the `AND event_json NOT LIKE '{"Name":%'` SQL fragment for every
+/// externally-tagged JSON discriminant in `prefixes`, newline-joined for
+/// readable queries. Used both to pin events against retention eviction and
+/// to exclude them from activity scans; the caller supplies the set so this
+/// module stays payload-agnostic.
+fn not_like_clauses(prefixes: &[&str]) -> String {
+    prefixes
+        .iter()
+        .map(|name| format!("AND event_json NOT LIKE '{{\"{name}\":%'"))
+        .collect::<Vec<_>>()
```
[the remark this answers is not in the document]
Comment (Seluj78) [reply src/events/mod.rs]: Done in https://github.com/agent-of-empires/agent-of-empires/pull/2374/commits/1979d4f7e — `not_like_clauses` now emits `NOT LIKE ?` placeholders and binds the patterns, so a future consumer's discriminant can't break or inject into the predicate. ACP's constants match identically.

File: src/events/mod.rs [hunk 2]
```diff
@@ -0,0 +1,678 @@
...
+    seq: u64,
+    json: &str,
+    created_at: i64,
+) -> Result<usize> {
+    let sql = format!(
+        "INSERT OR IGNORE INTO {} (session_id, seq, event_json, created_at)
+         VALUES (?1, ?2, ?3, ?4)",
+        schema.events_table()
+    );
+    conn.execute(&sql, params![topic, seq as i64, json, created_at])
```
[the remark this answers is not in the document]
Comment (Seluj78) [reply src/events/mod.rs:167]: Leaving as-is by design. These casts are preserved from the original ACP code; `seq`/`max_events` are app-assigned monotonic counters that never approach i64::MAX, and the one realistic u64::MAX sentinel (the scan cursor) is already clamped via `i64::try_from(..).unwrap_or(i64::MAX)` in `scan`. A checked conversion here would only change error behavior for an unreachable input.

File: src/events/mod.rs [hunk 3]
```diff
@@ -0,0 +1,678 @@
...
+    // pruned. A flat `seq <= cutoff` delete cannot strand a blob whose
+    // event survived, because attachments only ever hang off prunable
+    // events, never the pinned variants.
+    if let Err(e) = conn.execute(
+        &format!(
+            "DELETE FROM {} WHERE session_id = ?1 AND seq <= ?2",
+            schema.attachments_table()
+        ),
+        params![topic, cutoff],
+    ) {
```
[the remark this answers is not in the document]
Comment (Seluj78) [reply src/events/mod.rs]: Done in https://github.com/agent-of-empires/agent-of-empires/pull/2374/commits/1979d4f7e, refined in https://github.com/agent-of-empires/agent-of-empires/pull/2374/commits/9b242e0c. The attachment prune is now tied to event existence (`seq <= cutoff AND seq NOT IN (SELECT seq FROM events)`) rather than a flat `seq <= cutoff`, so a pinned event that survives the prune keeps its blobs. Identical result for ACP; correct for any future consumer.

File: src/process/worker.rs [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+/// on a shared host cannot enumerate worker ids.
+pub fn ensure_dir(dir: &Path) -> Result<()> {
+    if !dir.exists() {
+        std::fs::create_dir_all(dir)
+            .with_context(|| format!("creating worker dir at {}", dir.display()))?;
+        #[cfg(unix)]
+        {
+            use std::os::unix::fs::PermissionsExt;
+            let _ = std::fs::set_permissions(dir, std::fs::Permissions::from_mode(0o700));
+        }
```
[the remark this answers is not in the document]
Comment (Seluj78) [reply src/process/worker.rs]: Done in https://github.com/agent-of-empires/agent-of-empires/pull/2374/commits/d03a10ba — `ensure_dir` now (re)applies 0700 on every call, including a pre-existing directory, and propagates a `set_permissions` failure instead of swallowing it, so callers fail closed rather than proceeding with an enumerable worker dir.

File: src/process/worker.rs [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use tempfile::TempDir;
+
+    #[test]
+    fn is_pid_alive_self() {
+        assert!(is_pid_alive(std::process::id()));
+    }
```
[the remark this answers is not in the document]
Comment (Seluj78) [reply src/process/worker.rs:227]: Done in https://github.com/agent-of-empires/agent-of-empires/pull/2374/commits/d03a10ba — gated `is_pid_alive_self` behind `#[cfg(unix)]`, matching the non-unix `is_pid_alive` fallback.

File: src/events/mod.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Seluj78) [reply src/events/mod.rs]: Good catch, fixed in https://github.com/agent-of-empires/agent-of-empires/pull/2374/commits/9b242e0c. The events are now deleted first with an early return on failure, and the attachment cleanup runs only afterward, dropping blobs whose owning event is no longer present. A failed event prune no longer strands surviving events without their blobs.

**Review by njbrake — APPROVED** [review]
